File: nodes/core/io/text_input_node.py
from typing import Any

from nodes.base.base_node import Base


class TextInput(Base):
    """Simple node that outputs a static text value from parameters."""

    inputs = {}
    outputs = {"text": str}
    # Support both legacy "value" and preferred "text" parameter keys
    default_params = {"value": "", "text": None}
    params_meta = [{"name": "value", "type": "textarea", "default": ""}]

    async def _execute_impl(self, inputs: dict[str, Any]) -> dict[str, Any]:
        # Prefer explicit "text" param if provided; fall back to legacy "value"
        value = self.params.get("text")
        if value is None or (isinstance(value, str) and value == ""):
            value = self.params.get("value", "")
        return {"text": value}

File: backend/src/nodes/core/io/text-input-node.ts
// src/nodes/core/io/text-input-node.ts
// Translated from: nodes/core/io/text_input_node.py

import { Base } from '../../base/base-node';
import type { NodeInputs, NodeOutputs, ParamMeta, DefaultParams } from '../../../core/types';

/**
 * Simple node that outputs a static text value from parameters.
 */
export class TextInput extends Base {
  static override inputs: Record<string, unknown> = {};
  static override outputs: Record<string, unknown> = { text: String };

  // Support both legacy "value" and preferred "text" parameter keys
  static override defaultParams: DefaultParams = { value: '', text: null };
  static override paramsMeta: ParamMeta[] = [
    { name: 'value', type: 'textarea', default: '' },
  ];

  protected override async executeImpl(inputs: NodeInputs): Promise<NodeOutputs> {
    // Prefer explicit "text" param if provided; fall back to legacy "value"
    let value = this.params.text;
    if (value === null || value === undefined || (typeof value === 'string' && value === '')) {
      value = this.params.value ?? '';
    }
    return { text: value };
  }
}
